optuna/progress_bar.py

```diff
@@ -75,15 +75,25 @@ def _init_valid(self) -> None:
         optuna_logging.disable_default_handler()
         optuna_logging._get_library_root_logger().addHandler(_tqdm_handler)
 
-    def update(self, elapsed_seconds: float) -> None:
+    def update(self, elapsed_seconds: float, best_trial: int, best_value: Optional[float]) -> None:
         """Update the progress bars if ``is_valid`` is :obj:`True`.
 
         Args:
             elapsed_seconds:
                 The time past since :func:`~optuna.study.Study.optimize` started.
+            best_trial:
+                The number of the current best trial.
+            best_value:
+                The score of the current best trial.
         """
 
         if self._is_valid:
+            if self._n_trials is None and self._timeout is None:
+                assert False
```

Comment:
is this refactoring?

Reply:
Yeah. It's not necessary anymore, since I applied you suggestions and got rid of the custom description for now.